torchtext/experimental/transforms.py

```diff
@@ -251,3 +251,57 @@ def to_ivalue(self):
             vector = self.vector.to_ivalue()
             return VectorTransform(vector)
         return self
+
+
+class PadTransform(nn.Module):
+    r"""Pad sequences. Add padding id to the end of a sequences such that the sequences will
+        have the same length.
+
+    Args:
+        pad_id: the id for padding token
+        eos_token_id: the token id for the end-of-sentence mark (Default: None)
+        return_key_padding_mask: flag to output key_padding_mask as output (Defualt: True)
+
+    Example:
+        >>> pad_id = 2
+        >>> pad_transform = PadTransform(pad_id)
+        >>> seq_batch = [[5, 4, 5, 6, 7], [1, 3], [7, 5, 8]]
+        >>> pad_seq, key_padding_mask = pad_transform(seq_batch)
+        >>> jit_pad_transform = torch.jit.script(pad_transform)
+    """
+
+    def __init__(self, pad_id, eos_token_id=None, return_key_padding_mask=True):
+        super(PadTransform, self).__init__()
+        self.pad_id = pad_id
+        self.eos_token_id = eos_token_id
+        self.return_key_padding_mask = return_key_padding_mask
+
+    @torch.jit.export
+    def forward(self, seq_batch: List[List[int]]) -> Tuple[torch.Tensor, Optional[Tensor]]:
+        r"""Pad a list of integer lists. The individual integer list might have different length
+        such that the padding function will add padding id to the end of list for the same length.
+
+        Args:
+            seq_batch: a list of integer lists. Type: List[List[int]]
+
+        Outputs:
+            padded_sequence, key_padding_mask Type: Tuple[torch.Tensor, Optional[Tensor]]
```

Comment:
key_padding_mask is MHA specific. Is that necessary?

Reply:
fixed.